src/System.IO.FileSystem/tests/Directory/Regressions.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class Regressions : FileSystemTest
+    {
+        [Fact]
+        public void FileEnumeratorIsThreadSafe()
+        {
+            string directory = Directory.CreateDirectory(GetTestFilePath()).FullName;
+            for (int i = 0; i < 100; i++)
+                File.Create(Path.Combine(directory, GetTestFileName())).Dispose();
+
+            new ThreadSafeRepro().Execute(directory);
+        }
+
+        class ThreadSafeRepro
+        {
+            volatile IEnumerator<string> _enumerator;
+
+            void Race(IEnumerator<string> s)
```

Comment:
Nit: Could you please rename `Race` to something else? The world tends to get flagged by policheck...

Reply:
Done